src/Model/BaseManager.php

```diff
@@ -66,22 +64,22 @@ public function getObjectManager()
         return $manager;
     }
 
-    public function getClass()
+    public function getClass(): string
     {
         return $this->class;
     }
 
-    public function findAll()
+    public function findAll(): array
```

Comment:
Not a specialist of templates but maybe we could use phpdoc here to document this is `array<T>`?, and have we have `class-string<T>`?

Reply:
Thank you for the feedback. I'm in the same situation than you about generics, but I'd try to add the docblocks :+1:

Reply:
Looking at `ManagerInterface`, I see the generics are declared there:
https://github.com/sonata-project/sonata-doctrine-extensions/blob/f227410bdc1ea11909f943ec8e6ab46b135ac5f3/src/Model/ManagerInterface.php#L32-L37

Maybe they are just inherited here and there is no need to duplicate them.
I'll let this PR as is by now.